src/darwin/Framework/CHIP/MTRDeviceOverXPC.m

```diff
@@ -28,18 +29,21 @@
 
 @interface MTRDeviceOverXPC ()
 
-@property (nonatomic, strong, readonly) id<NSCopying> controller;
+@property (nonatomic, strong, readonly, nullable) id<NSCopying> controllerID;
+@property (nonatomic, strong, readonly) MTRDeviceControllerOverXPC * controller;
 @property (nonatomic, readonly) NSNumber * nodeID;
 @property (nonatomic, strong, readonly) MTRDeviceControllerXPCConnection * xpcConnection;
 
 @end
 
 @implementation MTRDeviceOverXPC
 
-- (instancetype)initWithController:(id<NSCopying>)controller
-                          deviceID:(NSNumber *)deviceID
-                     xpcConnection:(MTRDeviceControllerXPCConnection *)xpcConnection
+- (instancetype)initWithControllerID:(id<NSCopying> _Nullable)controllerID
+                          controller:(MTRDeviceControllerOverXPC *)controller
+                            deviceID:(NSNumber *)deviceID
+                       xpcConnection:(MTRDeviceControllerXPCConnection *)xpcConnection
```

Comment:
The fact that both `controllerID` and `controller` needs to be passed in is very confusing IMHO. Especially given that the underlying code is calling method ending with "Controller" but use "controllerId" as an argument (`subscribeWithController`, `readAttributeWithController`, * ...)

Just curious why, can't we just pass the `MTRDeviceControllerOverXPC` and check the `controllerID` property directly (as exposed by `MTRDeviceControllerOverXPC_Internal.h`)

Reply:
I guess we could, sure.  So here "controller" would mean the actual MTRDeviceControllerOverXPC and in other places it would mean the controller id.  Maybe I should call this one `initWithControllerOverXPC` or something to make them look different?

Reply:
`initWithControllerOverXPC` looks good to me!

Reply:
Done.